lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -102,22 +102,17 @@ def server_version
       kubectl.server_version
     end
 
-    def initialize(namespace:, context:, current_sha:, template_dir:, logger:, kubectl_instance: nil, bindings: {},
-      max_watch_seconds: nil, selector: nil)
+    def initialize(namespace:, context:, current_sha:, logger:, kubectl_instance: nil, bindings: {},
+      max_watch_seconds: nil, selector: nil, template_dirs: [], template_dir: nil)
       @namespace = namespace
       @namespace_tags = []
       @context = context
       @current_sha = current_sha
-      @template_dir = File.expand_path(template_dir)
+      @template_dirs = (template_dirs.map { |dir| File.expand_path(dir) } << template_dir).compact
```

Comment:
We don't need to worry about the case of `template_dir` and `template_dirs` both existing when invoking via CLI, but I've done it this way to make it accommodate the case of someone using the API itself.